federation/executor.go

```diff
@@ -291,7 +323,7 @@ func (e *Executor) execute(ctx context.Context, p *Plan, keys []interface{}, opt
 				if _, ok := subPlanMetaData.results[k]; !ok {
 					subPlanMetaData.results[k] = v
 				} else {
-					if k != keyField || v != subPlanMetaData.results[k] {
+					if k != keyField && v == subPlanMetaData.results[k] {
```

Comment:
Why make this change?

Reply:
it was from the rebase :( changing it back